### PR TITLE
Filter Invalid upgrade and Wondex fix

### PR DESCRIPTION
**Pull request by 5of9**
"filter.invalid" filters now impossible positions. 
In Wondex protocol answers to messages are no longer marked as invalid, even without position.
Sat over 2 check removed.
Wondex decoder test changed to pass because of the removed Sat over 2 check

### CODE REVIEW

File: src/org/traccar/protocol/WondexProtocolDecoder.java [hunk 1]
```diff
@@ -107,7 +106,7 @@ protected Object decode(
             position.setAltitude(parser.nextDouble(0));
 
             int satellites = parser.nextInt(0);
```
Comment (tananaev) [inline src/org/traccar/protocol/WondexProtocolDecoder.java]: Please remove this variable. It's no longer needed.

File: src/org/traccar/FilterHandler.java [hunk 2]
```diff
@@ -84,7 +84,12 @@ public FilterHandler() {
     }
 
     private boolean filterInvalid(Position position) {
-        return filterInvalid && !position.getValid();
+        if (filterInvalid) {
+           return !position.getValid() || position.getLatitude() > 90
```
Comment (tananaev) [inline src/org/traccar/FilterHandler.java]: Indentation is completely wrong here on all 3 lines.

File: src/org/traccar/FilterHandler.java [hunk 3]
```diff
@@ -84,7 +84,12 @@ public FilterHandler() {
     }
 
     private boolean filterInvalid(Position position) {
-        return filterInvalid && !position.getValid();
```
Comment (tananaev) [inline src/org/traccar/FilterHandler.java]: I think you can still use return expression straight away. You just need to groups the rest in parentheses.

Also, I liked the split of latitude and longitude checks on separate lines as you had before. I think it makes it more readable.

**Comment by tananaev** [comment]
Merged, thanks.

**Comment by Abyss777** [comment]
Hello
Frankly I think it is not good idea to make all positions valid for wondex protocol. It is not true.
here is quote from documentation

> 1) The device returns the last valid GPS information upon request regardless the
GPS reception. The parameter of “Number of Satellites” is ‘0’ if there is no GPS
reception or GPS is not fixed. Thus the parameter of “number of satellite” could be
a reference to check whether there is GPS reception or not.

**Comment by tananaev** [comment]
That's true for many protocol. It returns outdated, but valid location. It's still valid, so should be marked as valid. Timestamp would indicate that it's outdated.

**Comment by Abyss777** [comment]
Unfortunately, wondex devices do not report fixTime separately.
Do we have any attribute to mark it "outdated" ?

**Comment by Abyss777** [comment]
Oh, sorry. We have ordinary field for this.

**Comment by tananaev** [comment]
That field is only for marking positions that didn't have any location data. Also, it's not saved in the database.

Are you sure time returned by the device is not "fix time"?

**Comment by Abyss777** [comment]
Yes, first position after morning wake up has previous coordinates, satellites = 0 and current time, not yesterday.

**Comment by tananaev** [comment]
OK, maybe we should check if satellites number is zero.

**Comment by Abyss777** [comment]
And mark such position as invalid?

**Comment by 5of9** [comment]
@Abyss777, will you change this, or should I do it? If you want to fix this we could also change jobs, I could fix this, because I would need help with the "duplicate.filter issue" I opened...

**Comment by Abyss777** [comment]
@5of9 It's already implemented and merged in #3258

**Pull request merged**